Assets/MixedRealityToolkit/MSBuild/Scripts/Utilities.cs

```diff
@@ -0,0 +1,345 @@
+        public static bool TryGetTextTemplate(string text, string templateName, out string template)
+        {
+            string regex = $"^(\\s*#{templateName}_TEMPLATE .*)$";
+            Match result = Regex.Match(text, regex, RegexOptions.Multiline);
+
+            if (result.Success && result.Groups[1].Success && result.Groups[1].Captures.Count > 0)
+            {
+                template = result.Groups[1].Captures[0].Value;
+                return true;
+            }
+
+            template = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Given a list of Asset guids converts them to asset paths in place.
+        /// </summary>
+        public static void GetPathsFromGuidsInPlace(string[] guids)
+        {
+            for (int i = 0; i < guids.Length; i++)
+            {
+                guids[i] = AssetDatabase.GUIDToAssetPath(guids[i]);
+            }
+        }
+
+        /// <summary>
+        /// Helper to see if the specified BuildTarget is installed in the editor.
+        /// </summary>
+        public static bool IsPlatformInstalled(BuildTarget buildTarget)
+        {
+#if UNITY_EDITOR
+            Type moduleManager = Type.GetType("UnityEditor.Modules.ModuleManager, UnityEditor.dll");
+            MethodInfo isPlatformSupportLoaded = moduleManager.GetMethod("IsPlatformSupportLoaded", BindingFlags.Static | BindingFlags.NonPublic);
+            MethodInfo getTargetStringFromBuildTarget = moduleManager.GetMethod("GetTargetStringFromBuildTarget", BindingFlags.Static | BindingFlags.NonPublic);
+
+            return (bool)isPlatformSupportLoaded.Invoke(null, new object[] { (string)getTargetStringFromBuildTarget.Invoke(null, new object[] { buildTarget }) });
+#else
+            throw new PlatformNotSupportedException($"{nameof(IsPlatformInstalled)} is only supported in the editor.");
+#endif
+        }
+
+        /// <summary>
+        /// Gets a <see cref="BuildTargetGroup"/> for a specified <see cref="BuildTarget"/>.
+        /// </summary>
+        public static BuildTargetGroup GetBuildTargetGroup(BuildTarget buildTarget)
+        {
+            switch (buildTarget)
+            {
+                case BuildTarget.iOS:
+                    return BuildTargetGroup.iOS;
+                case BuildTarget.Android:
+                    return BuildTargetGroup.Android;
+                case BuildTarget.StandaloneWindows:
+                case BuildTarget.StandaloneWindows64:
+                    return BuildTargetGroup.Standalone;
+                case BuildTarget.WSAPlayer:
+                    return BuildTargetGroup.WSA;
+                case BuildTarget.NoTarget:
+                    return BuildTargetGroup.Unknown;
+                default:
+                    throw new PlatformNotSupportedException($"Don't currently support {buildTarget}");
+            }
+        }
+
+        public static IEnumerable<Version> GetUWPSDKs()
+        {
+            // May also be of interest to GetReferences
```

Comment:
>GetReferences [](start = 42, length = 13)

GetReferences isn't defined in this class, and is fairly generically named - this comment makes me ask more questions than it answers, so it might be better to just delete this, or provide additional context to make it more clear